src/Rules/Functions/SortParameterCastableToStringRule.php

```diff
@@ -0,0 +1,149 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\ArgumentsNormalizer;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Rules\RuleLevelHelper;
+use PHPStan\Type\Constant\ConstantIntegerType;
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeUtils;
+use PHPStan\Type\VerbosityLevel;
+use function array_key_exists;
+use function in_array;
+use function sprintf;
+use const SORT_NUMERIC;
+use const SORT_REGULAR;
+
+/**
+ * @implements Rule<Node\Expr\FuncCall>
+ */
+class SortParameterCastableToStringRule implements Rule
+{
+
+	public function __construct(
+		private ReflectionProvider $reflectionProvider,
+		private RuleLevelHelper $ruleLevelHelper,
+	)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return FuncCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!($node->name instanceof Node\Name)) {
+			return [];
+		}
+
+		if (!$this->reflectionProvider->hasFunction($node->name, $scope)) {
+			return [];
+		}
+
+		$functionReflection = $this->reflectionProvider->getFunction($node->name, $scope);
+		$functionName = $functionReflection->getName();
+
+		if (!in_array($functionName, ['array_unique', 'sort', 'rsort', 'asort', 'arsort'], true)) {
+			return [];
+		}
+
+		$origArgs = $node->getArgs();
+		$parametersAcceptor = ParametersAcceptorSelector::selectFromArgs(
+			$scope,
+			$origArgs,
+			$functionReflection->getVariants(),
+			$functionReflection->getNamedArgumentsVariants(),
+		);
+
+		$errorMessage = 'Parameter %s of function %s expects an array of values castable to string, %s given.';
+		$functionParameters = $parametersAcceptor->getParameters();
+		$castFn = static fn (Type $t) => $t->toString();
+
+		$normalizedFuncCall = ArgumentsNormalizer::reorderFuncArguments($parametersAcceptor, $node);
+
+		if ($normalizedFuncCall === null) {
+			return [];
+		}
+
+		$normalizedArgs = $normalizedFuncCall->getArgs();
+		if (!array_key_exists(0, $normalizedArgs)) {
+			return [];
+		}
+
+		$argsToCheck = [0 => $normalizedArgs[0]];
+		$flags = null;
+		if (array_key_exists(1, $normalizedArgs)) {
+			$flags = $scope->getType($normalizedArgs[1]->value);
+		} elseif (array_key_exists(1, $functionParameters)) {
+			$flags = $functionParameters[1]->getDefaultValue();
+		}
+
+		if ($flags === null || $flags->equals(new ConstantIntegerType(SORT_REGULAR))) {
```

Comment:
should this check support union-types?

Reply:
All other valid flags are stricter, so IMO it's not necessary to support union types in this condition.